odoo/api.py

```diff
@@ -690,7 +690,22 @@ def lang(self):
 
         :rtype: str
         """
-        return self.context.get('lang') or None
+        lang = self.context.get('lang')
+        if lang and lang != 'en_US' and not self['res.lang']._lang_get_id(lang):
+            raise UserError(_('Invalid language code: %s', lang))
```

Comment:
Logically it shouldn't have UX problem.
In the master
Usually when client access with inactive language, the http stack will fallback to an active language first. 
And ideally you shouldn't fetch translated content before that.
then the language will be checked when `fetch/write/crate` translated fields

in `get_lang`,
1. it may read `res.lang` which don't have translated field => safe
2. it uses env.context.get('lang') instead of env.lang => safe

so if we find the problem in web client access, we should fix it in http stack like task https://github.com/odoo/odoo/pull/151502
The current language check mainly works for rpc which prevent users reading unexpected 'en_US' languages

----

But I also discussed with @Julien00859 before, we think 
1. checking language when fetch may expose some coding problem.
And thanks to that he created task-3721341 to solve the context consitency issue in http
2. But the language check may introduce some unnecesary error as you said for the client user.

I wanted to remove this check for read after the task for juc, but he told me it would be cancelled yesterday. So I will remove it in this task.